master/internal/api_checkpoint.go

```diff
@@ -28,6 +35,82 @@ func (a *apiServer) GetCheckpoint(
 	}
 }
 
+func (a *apiServer) DeleteCheckpoints(
+	ctx context.Context,
+	req *apiv1.DeleteCheckpointsRequest) (*apiv1.DeleteCheckpointsResponse, error) {
+	curUser, _, err := grpcutil.GetUser(ctx, a.m.db, &a.m.config.InternalConfig.ExternalSessions)
+	if err != nil {
+		return nil, err
+	}
+
+	checkpointsToDeleteStr := req.CheckpointUuids
+
+	conv := &protoconverter.ProtoConverter{}
+	checkpointsToDelete := conv.ToUUIDList(checkpointsToDeleteStr)
+	if cErr := conv.Error(); err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "converting checkpoint: %s", cErr)
+	}
+
+	registeredCheckpointUUIDs, err := a.m.db.FilterForRegisteredCheckpoints(checkpointsToDelete)
+	if err != nil {
+		return nil, err
+	}
+
+	// return 400 if model registry checkpoints and include all the model registry checkpoints
+	if len(registeredCheckpointUUIDs) > 0 {
+		return nil, status.Errorf(codes.InvalidArgument,
+			"This subset of checkpoints provided are in the model registry and cannot be deleted: %v.",
+			registeredCheckpointUUIDs)
+	}
+
+	addr := actor.Addr(fmt.Sprintf("checkpoints-gc-%s", uuid.New().String()))
+
+	taskSpec := *a.m.taskSpec
+	agentUserGroup, err := a.m.db.AgentUserGroup(curUser.ID)
+	if err != nil {
+		return nil, status.Errorf(
+			codes.InvalidArgument,
+			"cannot find user and group information for user %s: %s",
+			curUser.Username,
+			err,
+		)
+	}
+	if agentUserGroup == nil {
+		agentUserGroup = &a.m.config.Security.DefaultTask
+	}
+
+	jobID := model.NewJobID()
+	if err = a.m.db.AddJob(&model.Job{
+		JobID:   jobID,
+		JobType: model.JobTypeCheckpointGC,
+		OwnerID: &curUser.ID,
+	}); err != nil {
+		return nil, fmt.Errorf("persisting new job: %w", err)
+	}
+
+	groupCUUIDsByEIDs, err := a.m.db.GroupCheckpointUUIDsByExperimentID(checkpointsToDelete)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, expIDcUUIDs := range groupCUUIDsByEIDs {
+		exp, eErr := a.m.db.ExperimentByID(expIDcUUIDs.ExperimentID)
+		if eErr != nil {
+			return nil, err
+		}
+
+		jobSubmissionTime := time.Now().UTC().Truncate(time.Millisecond)
+		ckptGCTask := newCheckpointGCTask(a.m, jobID, jobSubmissionTime, taskSpec,
+			exp.ID, exp.Config.AsLegacy(), expIDcUUIDs.CheckpointUUIDSStr, agentUserGroup, curUser)
+		if gcErr := a.m.system.MustActorOf(addr,
```

Comment:
`gcErr` isn't an error, it's an actor ref that should never be nil (else we would panic, hence the "must" in the name). so i'm suprisied this doesn't always error?

Reply:
I'm not sure about this. I will look into it. Even in api_experiment.go (line 215) it doesn't seem to error?

Reply:
oh wait, duh, it ends with .AwaitTermination

Reply:
if you notice, the one I wrote is in a goroutine and we await the actor's successful termination. in this one I think we just block the api call?

Reply:
Yes that makes sense.

Reply:
sent message in dmg but I think we still need to drop the AwaitTermination() here